main.js

```diff
@@ -0,0 +1,117 @@
+function setStorage() {
+  localStorage.setItem('books_data', JSON.stringify(BOOKS_DATA));
+}
+function retrieveStorage() {
+  BOOKS_DATA = JSON.parse(localStorage.getItem('books_data'));
+}
+// let BOOKS_DATA = JSON.parse(localStorage.getItem("books_data")) || [];
+if (!localStorage.getItem('books_data')) {
+  setStorage();
+} else {
+  retrieveStorage();
+}
+
+let form = document.querySelector('form');
+let booklist = document.getElementById('book-list');
+
+function updateSelections() {
+  form = document.querySelector('form');
+  booklist = document.getElementById('book-list');
+}
+
+function addBook() {
+  const bookTitle = document.getElementById('book-title');
+  const bookAuthor = document.getElementById('book-author');
+  const bookObj = {};
+
+  let exists = false;
+
+  for (let i = 0; i < BOOKS_DATA.length; i += 1) {
+    if (
+      BOOKS_DATA[i].title === bookTitle.value
+      && BOOKS_DATA[i].author === bookAuthor.value
+    ) {
+      exists = true;
+    }
+  }
+  if (!exists) {
+    bookObj.title = bookTitle.value;
+    bookObj.author = bookAuthor.value;
+    return bookObj;
+  }
+  return null;
+}
+
+function removeBook(element) {
+  const textElements = element.parentElement.querySelectorAll('.book-card-text');
+
+  BOOKS_DATA = BOOKS_DATA.filter((obj) => {
+    if (
+      textElements[0].innerText === obj.title
+      && textElements[1].innerText === obj.author
+    ) {
+      return false;
+    }
+
+    return true;
+  });
+}
+
+function generateBookCard(bookObj) {
+  const bookCard = document.createElement('li');
+  const bookTitle = document.createElement('p');
+  const bookAuthor = document.createElement('p');
+  const removeBtn = document.createElement('button');
+
+  bookCard.classList.add('book-card', 'mb-1');
+  bookTitle.classList.add('book-card-text');
+  bookAuthor.classList.add('book-card-text');
+  removeBtn.classList.add('remove-book', 'mb-05');
+  const { title, author } = bookObj;
+  bookTitle.textContent = title;
+  bookAuthor.textContent = author;
+
+  removeBtn.textContent = 'Remove';
+
+  bookCard.appendChild(bookTitle);
+  bookCard.appendChild(bookAuthor);
+  bookCard.appendChild(removeBtn);
+  return bookCard;
+}
+
+function clearBookSection() {
+  booklist.innerHTML = '';
+}
+
+function displayBookSection(bookArray) {
+  for (let i = 0; i < bookArray.length; i += 1) {
+    const bookCard = generateBookCard(bookArray[i]);
+    booklist.appendChild(bookCard);
+  }
+}
+
+displayBookSection(BOOKS_DATA);
+updateSelections();
+
+document.addEventListener('click', (event) => {
+  if (event.target && event.target.className === 'remove-book mb-05') {
+    removeBook(event.target);
+
+    setStorage();
+    clearBookSection();
+    displayBookSection(BOOKS_DATA);
+    updateSelections();
+  }
+});
+
+form.addEventListener('submit', () => {
+  if (addBook) {
+    BOOKS_DATA.push(addBook());
+  }
+
+  setStorage();
+  clearBookSection();
+  displayBookSection(BOOKS_DATA);
+  updateSelections();
+});
```

Comment:
[OPTIONAL] Please use `event.preventDefault()` so the browser will be not refreshing when you submit the form :slightly_smiling_face: 

```suggestion
form.addEventListener('submit', (e) => {
  e.preventDefault();
  if (addBook) {
    BOOKS_DATA.push(addBook());
  }

  setStorage();
  clearBookSection();
  displayBookSection(BOOKS_DATA);
  updateSelections();
});
```